lib/config.js

```diff
@@ -12,22 +12,60 @@
 import React from 'react'
 
 // Import the base icon set use/customize.
-import { ClassicModeIcon } from '@opentripplanner/icons'
+import { TriMetLegIcon, TriMetModeIcon } from '@opentripplanner/icons'
+
+/**
+ * This is a private helper function that can be used to determine the icon name
+ * to use based on information contained in the leg.
+ */
+function getIconNameForLeg (leg) {
+  if (
+    leg.routeLongName &&
+    leg.routeLongName.startsWith('Portland Streetcar')
+  ) {
+    return 'STREETCAR'
+  } else if (leg.rentedBike) {
+    return 'BIKETOWN'
+  }
+  return null
+}
 
 /**
  * @param  {Object} otpConfig The contents of the yml config file as json.
+ * @param  {Object} customIcons Mapping of mode to custom icon component
  * @return  This function must return an object and can have one of the
- * following functions:
- * - applyCustomIconLogic
- * - getItineraryFooter
+ * following:
+ * - getItineraryFooter (function)
+ * - LegIcon (component)
+ * - ModeIcon (component)
  */
-export function configure(otpConfig) {
+export function configure (otpConfig, customIcons = {}) {
+  // Build a ModeIcon component from a base ModeIcon component
+  // and from the customIcons set (they can't be imported from config.js).
+  // If there is an exact-matching icon in customIcons, use it.
+  const ModeIcon = ({ mode, ...props }) => {
+    // FIXME: Shouldn't this be returning Ferry for everything?
+    return <ClassicFerry mode={mode} {...props} />
+    if (!mode) return null;
+
+    const CustomIcon = customIcons[mode]
+    if (CustomIcon) return React.cloneElement(CustomIcon, {...props})
+
+    return <TriMetModeIcon mode={mode} {...props} />
+  }
+
+  // Build the LegIcon component, including custom icons for some legs if any.
+  const LegIcon = ({ leg, ...props }) => {
```

Comment:
For TriMet, simply write:
```js
const LegIcon = TriMetLegIcon
```
The TriMet customizations by leg are already in TriMetLegIcon.
Leave the customization example in as a comment.